plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -126,22 +147,78 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
+                        if (mlStatsInput.retrieveStatsForModel(modelId)) {
+                            Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
+                            for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
+                                if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
+                                    actionStatsMap.put(entry.getKey(), entry.getValue());
+                                }
+                            }
+                            modelStats.put(modelId, new MLModelStats(actionStatsMap));
                         }
                     }
-                    modelStats.put(modelId, new MLModelStats(actionStatsMap));
+                    // Count down the latch after each asynchronous call completes
+                    latch.countDown();
+                }, e -> {
+                    // Handle failure case here
+                    // For example, log the error
+                    log.error("Failed to retrieve model details for model ID: " + modelId, e);
+                    // Count down the latch even in case of failure to ensure proper synchronization
+                    latch.countDown();
+                }));
+            }
+            // Wait for all asynchronous calls to complete
+            try {
+                long timeoutInSeconds = 120; // Default timeout: 120 seconds
+                boolean allCompleted = latch.await(timeoutInSeconds, TimeUnit.SECONDS);
+
+                if (!allCompleted) {
+                    // Handle the case where not all calls completed within the timeout
+                    log.warn("Not all asynchronous calls completed within the specified timeout of 120 seconds");
                 }
+            } catch (InterruptedException e) {
+                // Handle interruption if necessary
+                Thread.currentThread().interrupt();
             }
         }
-
         return new MLStatsNodeResponse(clusterService.localNode(), statValues, algorithmStats, modelStats);
     }
+
+    private void validateAccess(String modelId, ActionListener<Boolean> listener) {
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+        String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            mlModelManager.getModel(modelId, null, excludes, ActionListener.runBefore(ActionListener.wrap(mlModel -> {
+                Boolean isHidden = mlModel.getIsHidden();
+                if (isHidden != null && isHidden) {
+                    if (isSuperAdmin) {
+                        listener.onResponse(true);
+                    } else {
+                        listener.onResponse(false);
+                    }
+                } else {
+                    listener.onResponse(true);
```

Comment:
From this, we don't check if user have access to this model ?

Reply:
Seems all users have access unless the model is hidden and they are not a super admin. The method name `validateAccess` seems misleading, how about change to `validateModelStatsViewAccess`